### PR TITLE
Fixes BaseButton not triggering the pressed signal on "Button Press"…

### LINKED ISSUE #34935: Action Mode "Button Press" in BaseButton nodes acts like "Button Release"
**Godot version:**
3.2.beta5

**OS/device including version:**
Windows 10 64-bit

**Issue description:**
On BaseButton nodes, if you put their Action Mode to "Button Press", the pressed signal gets sent on button release, and not on button press.

**Steps to reproduce:**
Create a node that derives from Base Button, set its Action Mode to "Button Press", connect the pressed signal to a script that prints something. The value will get printed on button release, not on button press.

**Minimal reproduction project:**
[Godot 3.2 Button Problem.zip](https://github.com/godotengine/godot/files/4037351/Godot.3.2.Button.Problem.zip)

### PR DESCRIPTION
**Pull request by OverloadedOrama**
…" Action Mode

This should fix #34935

This is my first ever PR, and I'm a little inexperienced with Godot's source code and C++ in general, so please double check it before merging in case I broke anything by accident! It does seem to be working (and the issue fixed) on my end when I compiled it, though.

### CODE REVIEW

File: scene/gui/base_button.cpp [hunk 1]
```diff
@@ -165,7 +165,7 @@ void BaseButton::on_action_event(Ref<InputEvent> p_event) {
 				_pressed();
 			}
 		} else {
-			if (!p_event->is_pressed()) {
+                        if ((p_event->is_pressed() && action_mode == ACTION_MODE_BUTTON_PRESS) || (!p_event->is_pressed() && action_mode == ACTION_MODE_BUTTON_RELEASE)) {
```
Comment (YeldhamDev) [inline scene/gui/base_button.cpp]: Use tabs instead of spaces.

**Comment by YeldhamDev** [comment]
You'll need to squash your commits: https://docs.godotengine.org/en/3.1/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

**Comment by OverloadedOrama** [comment]
Sorry about the spaces - not sure what happened there. I edited the file with QT Creator. Also sorry for not using `git commit --amend` properly, I'm still relatively new at git. I think I managed to squash the commits now though!

**Review by volzhs — APPROVED** [review]
looks good to me

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**